examplePrograms/library_components_export.py

```diff
@@ -0,0 +1,129 @@
+from nexarClients.design.nexarDesignClient import NexarClient
+import os
+import json
+import csv
+
+# This is the API query, feel free to chop and change this to suit your use case.
+# We recommend building queries in an IDE such as BananaCakePop or Postman.
+# You can use BananaCakePop in a browser here: https://api.nexar.com/graphql/
+
+workspaceURL = "Your workspace URL here"
+
+graphql_query = '''query library ($workspaceURL: String!, $after: String) {
+  desLibrary (workspaceUrl: $workspaceURL) {
+    components (first: 100, after: $after) {
+      totalCount
+      pageInfo {
+        hasNextPage
+        endCursor
+      }
+      nodes {
+        id
+        folder {
+          name
+        }
+        description
+        name
+        comment
+        details {
+          parameters {
+            name
+            value
+          }
+        }
+        manufacturerParts {
+          partNumber
+          companyName
+        }
+      }
+    }
+  }
+}'''
+
+
+if __name__ == "__main__":
+
+    # You will need to set your Nexar application's client ID and secret in your environment variables.
+    # For this example, the application should have the design scope enabled.
+    # Alternatively, switch how these are inputted but be careful not to share your credentials.
+
+    client_id = os.environ["NEXAR_CLIENT_ID"]
+    client_secret = os.environ["NEXAR_CLIENT_SECRET"]
+
+    # This class provides the necessary functionality to fetch access tokens and query the API.
+    # The scopes provided allow us to query Altium 365 design data.
+
+    nexar = NexarClient(client_id, client_secret, ["design.domain", "openid", "user.access"])
+
+    def retrieve_components():
+
+        page_info = {"hasNextPage": True, "endCursor": None}
+        components = []
+
+        while page_info["hasNextPage"] is True:
+
+            variables = {
+                "workspaceURL": workspaceURL,
+                "after": page_info["endCursor"]
+            }
+            response = nexar.get_query(graphql_query, variables)
+            components += response["desLibrary"]["components"]["nodes"]
+            page_info = response["desLibrary"]["components"]["pageInfo"]
+            print("Retrieved", len(components), "components...")
+
+        print("Retrieved", len(components), "components.")
+        return components
+
+    def json_export(components):
+
+        with open("altium_library_components.json", "w") as json_file:
+
+            json.dump(components, json_file)
+
```

Comment:
It's a bit hard to see which code is part of which function here. I think Python functions normally have two blank lines at the end instead of one, and that might solve the issue.